items/src/main/scala/weco/api/items/services/ItemUpdateService.scala

```diff
@@ -1,62 +1,110 @@
+  ): Future[ItemsWithIndex] = {
+    val items = itemsWithIndex.map(_._1)
 
-  def updateItems(
-    work: Work.Visible[WorkState.Indexed]
-  ): Future[List[Item[IdState.Minted]]] = Future.sequence {
-    work.data.items.map {
-      case item @ Item(IdState.Identified(_, srcId, _), _, _, _)
-          if isSierraId(srcId) =>
-        refreshItem(srcId, item)
-      case item => Future(item)
+    updateFunction(items).map { updatedItems =>
+      // Construct a lookup from SourceIdentifier -> index
+      val updatedItemsWithIndex = itemsWithIndex
+        .map {
+          case (item, index) => getSrcId(item) -> index
+        }
+        .flatMap {
+          // Add the correct index for an item by SourceIdentifier
+          case (srcId, index) =>
+            updatedItems.find(_.id.sourceIdentifier == srcId).map {
+              updatedItem =>
+                (updatedItem, index)
+            }
+        }
+
+      // Ensure that the update function has updated the correct number of results
+      require(
+        updatedItemsWithIndex.size == itemsWithIndex.size,
+        "Inconsistent results updating items: " +
+          s"Received: ${itemsWithIndex}, updated: ${updatedItemsWithIndex}"
+      )
+
+      updatedItemsWithIndex
     }
   }
+
+  /** Updates the Identified items on a work
+    *
+    *  Uses an ItemUpdater to update Identified items
+    *  where the ItemUpdater acts on a specific IdentifierType
+    *
+    *  @param work
+    *  @return a sequence of updated items
+    */
+  def updateItems(
+    work: Work.Visible[WorkState.Indexed]
+  ): Future[Seq[Item[IdState.Minted]]] =
+    Future.sequence {
+      // Group our results by IdentifierType or where Unidentifiable
+      work.data.items.zipWithIndex.groupBy {
+        case (Item(IdState.Identified(_, srcId, _), _, _, _), _) =>
+          Some(srcId.identifierType)
+        case (Item(IdState.Unidentifiable, _, _, _), _) => None
+      } map {
+        /* To satisfy the compiler when calling preserveOrder we need to add a type for itemsWithIndex
+         * The above groupBy ensures ONLY IdState.Identified is handed here, unknown to the compiler
+         * A further complication of type erasure means that we must acknowledge that Seq[_] is unchecked
+         */
+        case (
+            Some(identifierType),
+            itemsWithIndex: Seq[(Item[IdState.Identified], Int) @unchecked]
+            ) =>
+          // Look for an ItemUpdater for this IdentifierType and update
+          itemUpdatesMap
+            .get(identifierType)
+            .map(
+              updater =>
+                preservedOrderItemsUpdate(
+                  itemsWithIndex = itemsWithIndex,
+                  updateFunction = updater.updateItems
+                )
+            )
+            .getOrElse(Future(itemsWithIndex))
+        case (None, itemsWithIndex) =>
+          Future(itemsWithIndex)
+      }
+      // unzipWithIndex
+    } map (_.flatten.toList.sortBy(_._2).map(_._1))
```

Comment:
Would be nice to use named variables rather than `_2` and `_1` here.

Reply:
tuples with two things should have better names for accessing their elements :( I wonder if there is a way to do this without wrapping in a case class.

Reply:
The answer is: deconstruct the tuples and name them:

```scala
case (thingOne, _) => thingOne
```